## 1. 데이터셋 개요

제가 선정한 데이터는 포르투갈 고등학생들의 학업 및 생활 습관이 담긴 Kaggle 데이터 셋입니다. 비록 고등학생 대상의 데이터지만, '음주량', '연애 여부', '외출 빈도' 등 라이프스타일을 나타내는 변수들이 디테일하게 포함되어 있어 우리나라 대학생들의 일상적인 행동 패턴과 크게 다르지 않다고 보았습니다.

원래 이 데이터 셋의 분석 목적은 '학업 성적'을 예측하는 것입니다. 하지만 우리 프로젝트의 결과를 청중과 공유하고 직접 참여를 유도하는 과정에서 '학점'이라는 민감한 개인정보를 요구하는 것은 큰 걸림돌이 될 것이라고 생각했습니다.

따라서 '연애 여부(romantic)'을 새로운 타겟 변수로 설정하고자 합니다. 학업 대신 라이프스타일을 관련 열들을 독립 변수로 활용하여, 현재 솔로인지 커플인지를 예측하는 이진 분류 모델을 만들어본다면 재밌게 진행할 수 있을 거 같습니다.

In [1]:
import pandas as pd

In [3]:
df= pd.read_csv('English.csv')

In [4]:
df.head()

,school,sex,age,address,famsize,Pstatus,Medu,Fedu,Mjob,Fjob,...,famrel,freetime,goout,Dalc,Walc,health,absences,G1,G2,G3
0,GP,F,18,U,GT3,A,4,4,at_home,teacher,...,4,3,4,1,1,3,6,5,6,6
1,GP,F,17,U,GT3,T,1,1,at_home,other,...,5,3,3,1,1,3,4,5,5,6
2,GP,F,15,U,LE3,T,1,1,at_home,other,...,4,3,2,2,3,3,10,7,8,10
3,GP,F,15,U,GT3,T,4,2,health,services,...,3,2,2,1,1,5,2,15,14,15
4,GP,F,16,U,GT3,T,3,3,other,other,...,4,3,2,1,2,5,4,6,10,10


In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 395 entries, 0 to 394
Data columns (total 33 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   school      395 non-null    object
 1   sex         395 non-null    object
 2   age         395 non-null    int64 
 3   address     395 non-null    object
 4   famsize     395 non-null    object
 5   Pstatus     395 non-null    object
 6   Medu        395 non-null    int64 
 7   Fedu        395 non-null    int64 
 8   Mjob        395 non-null    object
 9   Fjob        395 non-null    object
 10  reason      395 non-null    object
 11  guardian    395 non-null    object
 12  traveltime  395 non-null    int64 
 13  studytime   395 non-null    int64 
 14  failures    395 non-null    int64 
 15  schoolsup   395 non-null    object
 16  famsup      395 non-null    object
 17  paid        395 non-null    object
 18  activities  395 non-null    object
 19  nursery     395 non-null    object
 20  higher    

[원본 변수들에 대한 설명]
school: 소속 학교 ('GP': Gabriel Pereira, 'MS': Mousinho da Silveira) - GP는 도시에 있는 학교이고, MS는 주변 외곽에 위치해 교육 인프라 접근성이 GP가 더 높아 더 학구적이고 교육열이 높은 분위기로 사료됨 

sex: 학생의 성별 ('F': 여성, 'M': 남성)

age: 학생의 나이 (15세 ~ 22세)

address: 거주지 유형 ('U': 도시/Urban, 'R': 시골/Rural)

internet: 가정 내 인터넷 접속 가능 여부 (yes / no)
famsize: 가족 규모 ('LE3': 3명 이하, 'GT3': 3명 초과)

Pstatus: 부모님 동거 여부 ('T': 함께 거주/Together, 'A': 별거/Apart)

Medu: 어머니의 교육 수준 (0: 무학, 1: 초등교육, 2: 중등교육, 3: 고등교육, 4: 대학교 이상)

Fedu: 아버지의 교육 수준 (0~4, 위와 동일)

Mjob: 어머니의 직업 ('teacher': 교사, 'health': 의료계, 'services': 서비스직, 'at_home': 전업주부, 'other': 기타)

Fjob: 아버지의 직업 (위와 동일)

guardian: 학생의 주 보호자 ('mother', 'father', 'other')

famrel: 가족 관계의 원만함 정도 (1: 매우 나쁨 ~ 5: 매우 좋음)

reason: 현재 학교를 선택한 이유 ('home': 집과 가까워서, 'reputation': 학교 명성, 'course': 교육 과정 선호, 'other': 기타)

traveltime: 통학 시간 (1: 15분 미만, 2: 15~30분, 3: 30분~1시간, 4: 1시간 이상)

studytime: 주간 학습 시간 (1: 2시간 미만, 2: 2~5시간, 3: 5~10시간, 4: 10시간 이상)

failures: 과거 해당 과목의 낙제 횟수 (1~3: 실제 횟수, 4: 4회 이상)

schoolsup: 학교 차원의 추가 교육/학습 지원 여부 (yes / no)

famsup: 가족 차원의 학업 지원 여부 (yes / no)

paid: 해당 과목(수학/포르투갈어) 유료 사교육(과외, 학원 등) 여부 (yes / no)

nursery: 유치원(보육원) 이수 여부 (yes / no)

higher: 고등교육(대학교) 진학 희망 여부 (yes / no)

activities: 방과 후 특별활동 및 동아리 참여 여부 (yes / no)

romantic: 현재 연애 중인지 여부 (yes / no)

freetime: 방과 후 자유 시간 (1: 매우 적음 ~ 5: 매우 많음)

goout: 친구들과 외출하여 어울리는 빈도 (1: 매우 적음 ~ 5: 매우 많음)

Dalc: 주중(Workday) 알코올 소비량 (1: 매우 적음 ~ 5: 매우 많음)

Walc: 주말(Weekend) 알코올 소비량 (1: 매우 적음 ~ 5: 매우 많음)

health: 현재 건강 상태 (1: 매우 나쁨 ~ 5: 매우 좋음)

absences: 학교 결석 총 횟수 (0 ~ 93)

G1: 1학기 성적 (0 ~ 20점)

G2: 2학기 성적 (0 ~ 20점)

G3: 최종 학기 성적 (0 ~ 20점)

## 2. 분류 예측시 사용할 변수들
타겟 변수
- romantic(연애 여부)

사교 및 유흥 지표
1. goout(외출 빈도)
2. Walc(주말 음주량)
3. Dalc(주중 음주량)

시간 자원 지표
1. freetime(자유 시간)
2. studytime(학습 시간)
3. activities(방과후 활동)

+) 파생 변수 
1. 총 음주량(Dalc + Walc)
2. 주말 폭음 편중도(Walc - Dalc) : 평일에는 자제하다 주말에 많이 마실 수록 솔로일 확률이 높을 것으로 가정
3. 종합 인싸력 지수 (goout + activities)

### B.U.T

행의 수가 395개로 사료가 매우 부족하다는 단점이 있습니다.